Wrappers/Python/cil/framework/framework.py

```diff
@@ -208,28 +208,75 @@ def subset(self, dimensions=None, **kw):
             temp.set_labels(dimensions)
             return temp      
 
+    def __get_sliced_size(self, cut, arr_in_length):
+        if isinstance(cut,slice):
+            length = len(range(arr_in_length)[cut])
+        else:
+            if cut == 'centre':
+                length = 1
+            else:
+                try:
+                    length = len(cut)
+                except TypeError:
```

Comment:
well, if they pass 1.0 it could be interpreted as an `int`, if they pass a `tuple` that's really no different to a `list`.

You think I should just use `isinstance` as I do on slice?

I was considering whether I'm being too harsh on the slice check and should just look for the start/stop/step attributes `if all(hasattr(form, attr) for attr in ['start', 'stop'. 'step']))`

Reply:
I've kept the duck typing but fixed the logic, good spot!